### PR TITLE
Signup page

### PR DESCRIPTION
**Pull request by Brunettow**
Overflow errors are handled in sign-up page and login page,
Small change in strong password check messages (Messages are now started to be showed when password box is clicked)

### CODE REVIEW

**Comment by omerfaunal** [comment]
Logic in Auth files are not yet ready. Those tasks assigned to @defabdullah. We need to wait for his commits before merging.

File: project/FrontEnd/collaborative_science_platform/lib/screens/login_page.dart [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+      await Provider.of<Auth>(context, listen: false).login(emailController.text, passwordController.text);
+    } on NoUserFound {
+      setState(() {
+        error = true;
+        errorMessage = "User not found.";
+      });
+    } on WrongPasswordException {
+      setState(() {
+        error = true;
+        errorMessage = "Password is wrong.";
```
Comment (omerfaunal) [inline project/FrontEnd/collaborative_science_platform/lib/screens/login_page.dart:55]: Change this as Password or username is wrong.

File: project/FrontEnd/collaborative_science_platform/lib/screens/signup_page.dart [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+                                focusNode: nameFocusNode,
+                                hintText: 'Name',
+                                color: error && nameController.text.isEmpty
+                                    ? AppColors.dangerColor
+                                    : AppColors.primaryColor,
+                                obscureText: false,
+                                prefixIcon: const Icon(Icons.person),
+                                suffixIcon: null,
+                                height: 64.0,
+                                onChanged: null,
```
Comment (omerfaunal) [inline project/FrontEnd/collaborative_science_platform/lib/screens/signup_page.dart:148]: Add onchanged strategy as checking if all criteria are matched.

File: project/FrontEnd/collaborative_science_platform/lib/widgets/app_button.dart [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import 'package:collaborative_science_platform/utils/colors.dart';
+import 'package:flutter/material.dart';
+
+class AppButton extends StatelessWidget {
+  final void Function()? onTap;
+  final String text;
+  final double height;
+
```
Comment (omerfaunal) [inline project/FrontEnd/collaborative_science_platform/lib/widgets/app_button.dart:8]: Add bool isActive

File: project/FrontEnd/collaborative_science_platform/lib/widgets/app_button.dart [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    required this.text,
+    required this.height,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return ElevatedButton(
+      onPressed: onTap,
+      style: ElevatedButton.styleFrom(
+        backgroundColor: AppColors.primaryColor,
```
Comment (omerfaunal) [inline project/FrontEnd/collaborative_science_platform/lib/widgets/app_button.dart:21]: If button is not active make its color grey[600]

File: project/FrontEnd/collaborative_science_platform/lib/widgets/strong_password_checks.dart [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  });
+
+  static bool passedAllPasswordCriteria(String password) {
+    bool minLengthIsMet = password.length >= minPasswordLength;
+    bool atLeastOneLowerCaseIsPresent = RegExp(r'[a-z]').hasMatch(password);
+    bool atLeastOneUpperCaseIsPresent = RegExp(r'[A-Z]').hasMatch(password);
+    bool atLeastOneNumberIsPresent = RegExp(r'\d').hasMatch(password);
+    bool atLeastOneSpecialCharacterIsPresent = RegExp(r'[!@#$%^&*/()_\-+{}\[\]:;<>,.?~\\|]').hasMatch(password);
+    return minLengthIsMet && atLeastOneLowerCaseIsPresent && atLeastOneUpperCaseIsPresent &&
+        atLeastOneNumberIsPresent && atLeastOneSpecialCharacterIsPresent;
```
Comment (omerfaunal) [inline project/FrontEnd/collaborative_science_platform/lib/widgets/strong_password_checks.dart:20]: Check if confirm and normal password are matched

**Review by omerfaunal — CHANGES_REQUESTED** [review]

**Review by omerfaunal — APPROVED** [review]

**Review by mehmetSuzer — APPROVED** [review]
ı tested this code in my environment. It looks nice! There is no problem as far as I see.

**Pull request merged**